### DISCUSSION TITLE
exposing npm_package_version as environment variable

### DISCUSSION THREAD

**Discussion by varand-pez** [vercel/community#458] [Help] [answered]
Hello,

I can with no issue expose `npm_package_version` and use it in the app when developing locally, however in Vercel when I put this value in the Environment Variables it doesn't get converted to the version number that is in package.json and prints it as string literal.

Would love to know how can I do this as I couldn't find anything in the docs.

Thanks.

**Comment by Aichnerc** [comment]
Any update with that?

**Reply by varand-pez** [reply]
nope, I got rid of the env variable and added it to next.config

**Reply by acordiner92** [reply]
I found a workaround to this by overriding the build step and setting the version via node
REACT_APP_VERSION=$(node -pe 'require("./package.json").version') npm run build

**Reply by noice-noise** [reply]
Any updates for this one? 

I wish to have **multiple references**  from my package.json props (e.g. `$npm_package_version`, `$npm_package_name`) which work properly on my local builds but Vercel deployments outputs just the exact string (instead of `0.1.0`) unfortunately.

**Reply by varand-pez** [reply]
@noice-noise you have to export it from `next.config.js`
```
const packageJson = require("./package.json");

{
  ...
  env: {
    APP_VERSION: packageJson.version,
  },
}
```
and use it like `process.env.APP_VERSION` in the code.
Hope it helped.

**Reply by CYDVici0us** [reply]
I just came across this same issue, and I don't have a nextjs.config to do this.  Would be nice if Vercel extended the ENV to include some of the known [NPM variables](https://gist.github.com/moos/4635bda5b04dc8113d8ea7ee974cabc2).

**Comment by swarnava** [comment, marked as answer]
You can probably override the [build step](https://vercel.com/docs/concepts/deployments/build-step) and set the version via node

```
REACT_APP_VERSION=$(node -pe 'require("./package.json").version') npm run build
```

**Reply by CYDVici0us** [reply]
I just tried this, and REACT_APP_VERSION came back as undefined.  I'm using Next JS.

**Reply by swarnava** [reply]
oh. For next.js, you can export it to `next.config.js` and access it : https://github.com/underscoredotspace/purple/blob/main/next.config.js

**Reply by varand-pez** [reply]
@swarnava you get get ton of benefits from having next.config.js, you can easily add it and take advantage of the benefits
